### DISCUSSION TITLE
Implement useIdle hook

### DISCUSSION THREAD

**Discussion by achmurali** [mantinedev/mantine#164] [Q&A] [answered]
Hi rvital,

Have a query regarding useIdle hook. 
The trello board mentions this point `Hook receives time in ms after which idle status (return value) is true`.

Does this mean i should return a promise, how does the user check if the returned value is true or false after the supplied timeout is elapsed, could you please give me some more clarity on how this useIdle hook is going to be used by a user.

**Comment by rtivital** [comment, marked as answer]
Use react state to store value, hook definition should be:

```tsx
function useIdle(timeout: number): boolean;
```
